csrc/python_frontend/fusion_definition.cpp

```diff
@@ -455,40 +454,40 @@ std::vector<DistributedTensor> FusionDefinition::execute(
 
   // Convert `at::Tensor`s to `DistributedTensor`s.
   std::vector<DistributedTensor> out_dtensors;
-  out_dtensors.reserve(out_tensors.size());
   if (user_sched == nullptr) {
     FusionKernelRuntime* runtime =
         scheds->auto_gen_schedules->getMostRecentKernelRuntime();
     Fusion* fusion = runtime->fusionSegments()->completeFusion();
 
     int64_t tensor_index = 0;
-    for (Val* out_val : fusion->outputs()) {
+    for (auto out_val : fusion->outputs()) {
+      NVF_ERROR(
+          out_val->isA<TensorView>(),
+          "Non tensor outputs not supported currently due to lack of support of DistributedTensor in KernelArgumentHolder");
       auto* out_tv = out_val->as<TensorView>();
       if (fusion->getOutputAlias(out_tv).hide_output) {
         continue;
       }
-
-      const at::Tensor& out_tensor = out_tensors.at(tensor_index);
-      tensor_index++;
+      const at::Tensor& out_tensor = outputs[tensor_index++].as<at::Tensor>();
```

Comment:
```suggestion
      const at::Tensor& out_tensor = outputs.at(tensor_index++).as<at::Tensor>();
```
To avoid memory corruption, I prefer `at` which comes with a bound check unless I'm certain it's going to be in bounds.

Reply:
IIUC, this `outputs` is `KernelArgumentHolder`, which uses `std::vector::at` internally.